src/XAD/Complex.hpp

```diff
@@ -1375,6 +1375,17 @@ XAD_INLINE xad::FReal<T> norm(const complex<xad::FReal<T>>& x)
     return ::xad::detail::norm_impl(x);
 }
 
+// appleclang15 needs this overload for type paramed norm
+#if defined(__APPLE__) && defined(__clang__)
+#if defined(__apple_build_version__) && (__apple_build_version__ >= 15000000)
+template <class T>
+XAD_INLINE typename std::enable_if<!std::is_arithmetic<T>::value, T>::type norm(complex<T>& x)
```

Comment:
This is better:

```suggestion
XAD_INLINE typename std::enable_if<!xad::ExprTraits<T>::isExpr, T>::type norm(complex<T>& x)
```

The expression traits are meant for this purpose, and check if `T` is an XAD active type of some kind.

Reply:
I believe you meant:

`XAD_INLINE typename std::enable_if<xad::ExprTraits<T>::isExpr, T>::type norm(complex<T>& x)`
(note the `!` after `enable_if`)

With your suggested fix and the correction the build works correctly again